docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
+  * [Common classes](#common-classes)
+* [Implementation](#implementation)
+  * [Bookmark and Timetable modes feature](#bookmark-and-timetable-modes-feature-tys)
+  * [Add Module and Slot feature](#add-module-and-slot-feature-xing-rong)
+* [Documentation, logging, testing, configuration, dev-ops](#architecture)
+* [Appendix: Requirements](#architecture)
+  * [Product Scope](#architecture)
+  * [User Stories](#architecture)
+  * [Use cases](#architecture)
+  * [Non-Functional Requirements](#architecture)
+  * [Glossary](#glossary)
+* [Appendix: Instructions for manual testing](#faq)
+  * [Launch and shutdown](#command-summary)
+
+
+## **Getting Started**
+Refer to the setting up guide over [here](https://github.com/AY2021S1-CS2113T-W11-1/tp/releases).
+
+
+
+
+
+
+## **Design**
+
+### Architecture
+
+The figure below shows a high-level design for the architecture of Zoomaster. 
+![](https://github.com/TYS0n1/tp/blob/team-Branch2/docs/diagrams/archiveture2-01.png?raw=true)
+*<center/> Figure 1.1 Architecture diagram of Zoomaster </center> <br/></br>*
+
+Our Program can be split up into 8 components
+* Initialization
+* User Interface
+* Parser
+* Commands
+* Temp Lists
+* Storage
+* Local Files
+* Exceptions
+
+These components interact with each other as shown in Figure 1.1 to execute the functionalities of Zoomaster.
+
+### Initialization
+The diagram below shows a class-level diagram for Zoomaster. <br/></br>
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/initial.png)
+*<center/> Figure 1.2 Class diagram of Initialization </center> <br/></br>*
+
+**API**:`Zoomaster.java`
+
+The Initialization component is responsive for setting up Zoomaster for it to be used by users. 
+
+It consists of `Zoomaster`, `Ui`, `Storage`, `BookmarkList`, `Timetable` and `Module` classes.
+
+Its main roles are:
+* Retrieving bookmark, timetable and planner lists from storage if it exists.
+* Set the list of Modules for Zoomaster 
+* Initializes the User Interface object, as well as the 3 Storage  objects for `BookmarkList`, `Timetable` and `Planner`.
+
+### User Interface component
+
+![](images/UiClassDiagram.png)
+*<center/> Figure 1.3 Class diagram of User Interface </center> <br/></br>*
+
+**API**:`Ui.java`
+
+The UI component is responsible for all visual output to the User from the app. 
+
+The only class carrying out the component's function is the `Ui` class.
+
+It utilises the `Jansi` library to enable the usage of ANSI escape codes to format the console, allowing the app to output in colour. <br/>
+
+The UI also receives input from the User using a `Scanner` object. It returns the input as a String to the main function. <br/>
+
+In addition, the UI contains the different exception and error messages which can be displayed. When a particular exception is thrown (eg. **UNKNOWN_INPUT**), the corresponding function is called in UI to print out the error message (**printUnknownInputMessage()**). 
```

Comment:
Because of the lack of a section explaining the formatting conventions used, I'm unsure of the difference between text formatted as `keywords` and text **in bold**. Do you think this could be explained somewhere in case others have similar difficulty?